File: q01_1.py
N = 4

cnt = 0
for rock in range(N+1):
    for scissors in range(N+1 - rock):
        paper = N - rock - scissors
        all = [rock, scissors, paper]
        max_index = [i for i, x in enumerate(all) if x == max(all)]
        if len(max_index) == 1:
            cnt += 1

print(cnt)

File: q01_1.rb
N = 100

cnt = 0
0.upto(N) do |rock|
    0.upto(N - rock) do |scissors|
        paper = N - rock -scissors
        all = [rock, scissors, paper]
        cnt += 1 if all.count(all.max) == 1
    end
end

puts cnt
